controllers/defaultremediation_controller_test.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"context"
+	"github.com/medik8s/default-remediation-operator/api/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("Default Remediation CR", func() {
+	Context("Defaults", func() {
+		var underTest *v1alpha1.DefaultRemediation
+
+		BeforeEach(func() {
+			underTest = &v1alpha1.DefaultRemediation{
+				ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "default"},
+			}
+			err := k8sClient.Create(context.Background(), underTest)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(context.Background(), underTest)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("creating a resource", func() {
+			It("CR is namespace scoped", func() {
+				Expect(underTest.Namespace).To(Not(BeEmpty()))
+			})
+
+		})
+	})
+
+	Context("Reconciliation", func() {
+		var (
+			underTest      *v1alpha1.DefaultRemediation
+			objects        []runtime.Object
+			reconciler     DefaultRemediationReconciler
+			reconcileError error
+			getNHCError    error
+		)
+
+		JustBeforeEach(func() {
+			cl := fake.NewClientBuilder().WithRuntimeObjects(objects...).Build()
+			reconciler = DefaultRemediationReconciler{Client: cl, Log: controllerruntime.Log, Scheme: scheme.Scheme}
+			_, reconcileError = reconciler.Reconcile(
+				context.Background(),
+				controllerruntime.Request{NamespacedName: types.NamespacedName{Name: underTest.Name, Namespace: underTest.Namespace}})
+			getNHCError = reconciler.Get(
```

Comment:
`getNHCError` is a confusing name

Reply:
Indeed - missed that when copy pasted tests from NHC :frowning_face: